client/app/account/billing/autoRenew/billing-autoRenew.controller.js

```diff
@@ -672,20 +669,22 @@ angular.module("Billing.controllers").controller("Billing.controllers.AutoRenew"
                 $scope.servicesTypes.unshift(ALL_SERVICE_TYPES);
 
                 // sets the model of the type select input.
-                $scope.serviceTypeObject = _.find($scope.servicesTypes, {
+                $scope.serviceTypeObject.value = _.find($scope.servicesTypes, {
                     key: $scope.services.selectedType
                 });
 
-                if (!$scope.serviceTypeObject) {
-                    $scope.serviceTypeObject = ALL_SERVICE_TYPES;
+                if (!$scope.serviceTypeObject.value) {
+                    $scope.serviceTypeObject.value = ALL_SERVICE_TYPES;
                 }
             });
 
             const userCertificatesPromise = AutoRenew.getUserCertificates().then((results) => {
                 $scope.canDisableAllDomains = _.includes(results, "domains-batch-autorenew");
             });
 
-            $q.all([autorenewSettingsPromise, userPromise, paymentMeansPromise, renewAlignUrlPromise, userGuidePromise, serviceTypePromise, userCertificatesPromise]).finally(() => (initLoading = false));
+            $q.all([$scope.getServices($scope.count, $scope.offset), userPromise, paymentMeansPromise, renewAlignUrlPromise, userGuidePromise, serviceTypePromise, userCertificatesPromise])
+                .finally(() => $scope.nicRenew.getNicRenewParam())
```

Comment:
You could use a single `finally` method.

Reply:
No, I want the loading to stop:
* Even if `$scope.nicRenew.getNicRenewParam()` throws an exception
* After `$scope.nicRenew.getNicRenewParam()` is done

This can't be done with only one `finally`

Reply:
Could be addressed with just a `.then` right? 🌻

Reply:
I think `$scope.nicRenew.getNicRenewParam()` does require some `Promise` above to be successful to work so I think you're right !